### PR TITLE
Add Entra ID OAuthe2 token support

### PR DESCRIPTION
**Pull request by asaflevi-ms**
Add support for AAD token credentials

- [ ] a private preview
- [ ] a public preview
- [X] GA release

### CODE REVIEW

File: specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json [hunk 1]
```diff
@@ -37,9 +37,24 @@
...
+    {
+      "AADToken": [
+        "https://cognitiveservices.azure.com/.default"
+      ]
     }
   ],
   "securityDefinitions": {
+    "AADToken": {
+      "type": "oauth2",
+      "description": "The Azure Active Directory OAuth2 Flow",
```
Comment (vicancy) [inline specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json:50]: Azure Active Directory is now Microsoft Entra ID https://learn.microsoft.com/en-us/entra/fundamentals/how-to-rename-azure-ad

Comment (asaflevi-ms) [reply specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json:56]: @vicancy .
Is this what you meant ?

**Comment by asaflevi-ms** [comment]
@marclerwick, @vicancy , @catalinaperalta , @weidongxu-microsoft 
Can one of you review this PR?
If ok, please approve and merge.

CC: @koen-mertens

File: specification/ai/HealthInsights/HealthInsights.OpenAPI/service.tsp [hunk 2]
```diff
@@ -16,13 +16,18 @@ using TypeSpec.Versioning;
...
-@useAuth(AzureKey)
+@useAuth(AzureKey | EntraIdToken)
 namespace AzureHealthInsights;
 
 @doc("The secret key for your Azure Cognitive Services subscription.")
 model AzureKey
   is ApiKeyAuth<ApiKeyLocation.header, "Ocp-Apim-Subscription-Key">;
 
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Supress casing style for AAD"
+@doc("The Azure Active Directory OAuth2 Flow")
```
Comment (catalinaperalta) [inline specification/ai/HealthInsights/HealthInsights.OpenAPI/service.tsp:27]:
```suggestion
@doc("The Microsoft Entra Id Flow")
```

File: specification/ai/HealthInsights/HealthInsights.OpenAPI/service.tsp [hunk 3]
```diff
@@ -16,13 +16,18 @@ using TypeSpec.Versioning;
...
 )
-@useAuth(AzureKey)
+@useAuth(AzureKey | EntraIdToken)
 namespace AzureHealthInsights;
 
 @doc("The secret key for your Azure Cognitive Services subscription.")
 model AzureKey
   is ApiKeyAuth<ApiKeyLocation.header, "Ocp-Apim-Subscription-Key">;
 
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Supress casing style for AAD"
```
Comment (catalinaperalta) [inline specification/ai/HealthInsights/HealthInsights.OpenAPI/service.tsp:26]: seems like you dont need this suppression anymore?

**Review by weidongxu-microsoft — COMMENTED** [review]
Are we adding AAD support to an existing api-version (that already serving customer)?

**Comment by asaflevi-ms** [comment]
> Are we adding AAD support to an existing api-version (that already serving customer)?

@weidongxu-microsoft 
Actually, Cognitive Services support it.
We just need to show this in our spec so we can correctly generate SDKs.

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json [hunk 4]
```diff
@@ -2108,10 +2127,23 @@
...
+          },
+          "required": [
+            "response"
+          ]
         }
       },
       "required": [
-        "error"
+        "error",
+        "requestId"
```
Comment (weidongxu-microsoft) [inline specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json:2146]: I don't see where these changes comes from.

There is no update to typespec file about these in this PR?

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft Actually, Cognitive Services support it. We just need to show this in our spec so we can correctly generate SDKs.

OK. For a recent process change, you can add `BreakingChanges-Approved-Bugfix` if these is no change in backend, and the typespec/swagger change is to "fix" API definition.

Though you need to resolve the conflict first.

File: specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json (hunk 4 above)
Comment (asaflevi-ms) [reply specification/ai/data-plane/HealthInsights/stable/2024-04-01/openapi.json:2146]: @weidongxu-microsoft 
Maybe caused by new compiler version?

**Pull request merged**